src/runtime_src/core/tools/xbmgmt2/SubCmdReset.cpp

```diff
@@ -67,13 +67,20 @@ reset_device(const std::shared_ptr<xrt_core::device>& dev, xrt_core::query::rese
 SubCmdReset::SubCmdReset(bool _isHidden, bool _isDepricated, bool _isPreliminary)
     : SubCmd("reset", 
              "Resets the given device")
+    , m_device("")
+    , m_resetType("hot")
+    , m_help(false)
 {
   const std::string longDescription = "Resets the given device.";
   setLongDescription(longDescription);
   setExampleSyntax("");
   setIsHidden(_isHidden);
   setIsDeprecated(_isDepricated);
   setIsPreliminary(_isPreliminary);
+  m_commonOptions.add_options()
+    ("device,d", boost::program_options::value<decltype(m_device)>(&m_device), "The Bus:Device.Function (e.g., 0000:d8:00.0) device of interest.")
+    ("help", boost::program_options::bool_switch(&m_help), "Help to use this sub-command")
+    ;
 }
```

Comment:
```suggestion
  m_hiddenOptions.add_options()
    ("type,t", boost::program_options::value<decltype(m_resetType)>(&m_resetType)->notifier(supported), "The type of reset to perform. Types resets available:\n"
                                                                        "  kernel       - Kernel communication links\n" 
                                                                        "  ert          - Reset management processor\n"
                                                                        "  ecc          - Reset ecc memory\n"
                                                                        "  soft-kernel  - Reset soft kernel")
  ;
}
```
This is missing the hidden type parameter from the original SubCmdReset

Reply:
This is the supported function:
```
static void
supported(std::string resetType) {
  std::vector<std::string> vec { "user" };
  std::vector<std::string>::iterator it;
  it = std::find (vec.begin(), vec.end(), resetType); 
  if (it == vec.end()) {
    throw xrt_core::error(-ENODEV, "reset not supported");
  }
}
```